Use version for publish step in build.

codito opened the pull request.

Commit:
Use version for publish step in build.

smadala reviewed (approved):
build.sh need to update as well.

Commit:
Do not use cancellation token with timeout for launching test host.

It actually takes in consideration the thread switch time. When thread count is
high (say on a 12 core machine), launching test host may take more time than expected.

Add version for build.sh publish.
Fix a restore issue.

Commit:
Remove not needed code.

smadala commented on src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs:
```diff
@@ -141,7 +141,10 @@ public void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEve
                 {
                     var rawMessage = this.TryReceiveRawMessage();
-                    EqtTrace.Info("received message: {0}", rawMessage);
+                    if (EqtTrace.IsInfoEnabled)
```
I feel, this can be verbose.

smadala commented on src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs:
```diff
@@ -405,11 +403,5 @@ private string GetTestHostPath(string runtimeConfigDevPath, string depsFilePath,
         }
-
-        private CancellationTokenSource GetCancellationTokenSource()
-        {
```
Remove `Timeout` variable declaration to.

Commit:
PR comments.
